src/app/categories/state/entry-categories.service.ts

```diff
@@ -60,5 +59,8 @@ export class EntryCategoriesService {
           this.entryCategoriesStore.setError(true);
         }
       );
+    } else {
+      this.entryCategoriesStore.set(null);
```

Comment:
In the code above, if an entry doesn't have any categories, this method is called with an empty array. I think you should do the same here. The signature for the method is: `set(entities: SetEntities<EntityType>...` where EntityType is the generic, and `SetEntities` is `export declare type SetEntities<Entity> = Entity[] | Entities<Entity> | HashMap<Entity>`, which is a long-winded way of saying the method shouldn't be passed a `null`.

Digression: We mean to turn on null checking some day so the compiler would flag this, but there are currently too many exceptions, so we can't. But we want to avoid introducing new code passing null unless the signature explicitly allows it.

Reply:
This section was removed per Steve's suggestion.